src/praisonai-agents/praisonaiagents/agent/chat_mixin.py

```
@@ -803,36 +803,92 @@ def _chat_completion(self, messages, temperature=1.0, tools=None, stream=None, r
             
-            # Check for non-retryable errors (auth issues)
-            auth_indicators = ["401", "403", "authentication", "unauthorized", "invalid_api_key"]
+            # Get token counts for context-aware classification
+            prompt_tokens = 0
+            context_length = 0
+            try:
+                from ..context.tokens import estimate_messages_tokens
+                prompt_tokens = estimate_messages_tokens(messages)
+                context_length = prompt_tokens
+            except Exception:
+                pass  # Token estimation failed, continue without counts
```

Ruff (0.15.15)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)